Updated 'locale' and 'multi-edition' logic

mohammad-haque opened the pull request:
## Summary

Currently, we have:
* `Locale`: that we use to auto-detect initial edition based on user device locale. It is remotely switched controlled
* `Multi edition`: this flag we use to show hide `edition switcher` button and also to decide which text to show, for example Daily vs Edition. It is now controlled from debug settings

This PR changes the behavior to:
* `Locale`: all beta users (including debug build) will have it on, we have removed the previous `default_locale` remote switch from the app and introduced a new one to avoid any incidents. The new one called `auto_edition_detection`
* `Multi edition`: all beta users (including debug build) will see it, we added a remote switch so we can turn in on later for the general public. The new remote switch is `auto_edition_detection`


We are making this change so we can test multi-edition change for the beta users before we go to production.
 

[**Trello Card ->**](https://trello.com/c/OMvasp0M/1478-allow-the-edition-switch-to-be-available-to-beta-users-but-not-live-users)
[**Trello Card ->**](https://trello.com/c/Z5xDC6im/1479-deprecate-existing-locale-remote-switch)

## Test Plan
To test in beta: we can just test internal beta build
To test in prod: we have some unit test for validation and also we can test this on locally produced prod build from Android Studio or Xode.

mohammad-haque commented:
Thanks @arausuy for reviewing this PR, I will avoid merging this PR until we are ready to release this work to beta.

jimhunty commented:
@mohammad-haque Is this PR now required now we are removing the switch here: https://github.com/guardian/editions/pull/1449 ?

mohammad-haque commented:
> #1449

Since we have removed the edition switch we don't need this PR. But I replaced a remove-switch flag (`locale`) that might be useful. I can close this PR and raise a new one with remote switch replacement flag?

jimhunty commented:
@mohammad-haque Sounds good to me 👍

mohammad-haque commented:
This has been covered by this PR (https://github.com/guardian/editions/pull/1462)

Closing this PR
